src/openforms/submissions/models/submission.py

```diff
@@ -782,3 +788,13 @@ def registration_backend(self) -> FormRegistrationBackend | None:
             if (default_key := self.default_registration_backend_key)
             else None
         )
+
+    @property
+    def on_completion_task_ids(self) -> list[str]:
+        """Get the IDs of the tasks triggered when the submission was completed
+
+        Other tasks are triggered later once the submission is cosigned, a payment is made or a retry flow is triggered.
+        But to give feedback to the user we only need the first set of triggered tasks.
+        """
```

Comment:
I'm not sure I like the `.first()` in here based on PK ordering - is there a chance we can let the chain scheduling function return the appropriate instance that contains the task IDs and use that in the `_complete` endpoint instead? That would make it deterministic since you only ever consider the record that was created in the same (HTTP) transaction.

*... some time passes ...*

I've reviewed the rest of the code and this is not viable because the record is only completed after the transaction commits, so there's no way to pass this record back and include it in the `SubmissionStatus` thing.

Perhaps we can pass a flag `from_complete_endpoint` to the `SubmissionTaskMetadata` instance, and add a unique together condition based on this flag so that we can uniquely identify the tasks ID set to use for the `/api/v2/submissions/:uuid/status` endpoint?

Reply:
I added a flag saying which event triggered the chain. The submission is completed only once, so we can filter on the on_completion event